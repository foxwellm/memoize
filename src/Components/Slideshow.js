import React, { Component } from 'react';
import Card from './Card';
import { uid } from 'react-uid';

export default class Slideshow extends Component {

  render() {
    const { currentPage, setFavorite, currentIndex, nextMethod, prevMethod, currentMethods, favorites } = this.props;

    return (
      <div className="Slideshow">
        <div className={`cards-slider active-slide-${currentIndex}`}>
          {currentIndex !== 0 &&
            <div
              className="prev-btn"
              onClick={prevMethod}>
              <i className="fas fa-arrow-circle-left"></i></div>
          }
          <div className="cards-slider-wrapper" style={{
            'transform': `translateX(-${currentIndex * (100 / currentMethods.length)}%)`
          }}>
            {
              currentMethods.map((method, i) =>
                <Card method={method}
                  key={uid}
                  setFavorite={setFavorite}
                  favorites={favorites}
                  currentPage={currentPage}
                  cardNumber={i} />
              )
            }
          </div>
          {currentIndex !== currentMethods.length - 1 &&
            <div
              className="next-btn"
              onClick={() => nextMethod(currentMethods)}>
              <i className="fas fa-arrow-circle-right"></i>
            </div>
          }
        </div>
      </div>
    )
  }
}