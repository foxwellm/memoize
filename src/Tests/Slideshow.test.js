import React from 'react';
import Slideshow from '../Components/Slideshow';
import { shallow } from 'enzyme';
import data from '../Data/index';

describe('Slideshow', () => {
  let wrapper;
  const currentPage = 'string';
  const stringMethods = data.stringMethods;
  const currentIndex = 0;
  const favorites = { string: ["concat", "charCodeAt"], array: ["every", "forEach"] }
  const setFavoriteMock = jest.fn();
  const nextMethodMock = jest.fn();
  const prevMethodMock = jest.fn();


  beforeEach(() => {

    wrapper = shallow(
      <Slideshow
        currentMethods={[`${currentPage}Methods`]}
        favorites={favorites[`${currentPage}`] || []}
        setFavorite={setFavoriteMock}
        currentPage={currentPage}
        currentIndex={currentIndex}
        nextMethod={nextMethodMock}
        prevMethod={prevMethodMock}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})