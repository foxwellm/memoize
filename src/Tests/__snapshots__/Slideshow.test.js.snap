// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideshow should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slideshow
    currentIndex={0}
    currentMethods={
      Array [
        "stringMethods",
      ]
    }
    currentPage="string"
    favorites={
      Array [
        "concat",
        "charCodeAt",
      ]
    }
    nextMethod={[MockFunction]}
    prevMethod={[MockFunction]}
    setFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cards-slider active-slide-0"
      >
        <div
          className="cards-slider-wrapper"
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <Card
            cardNumber={0}
            currentPage="string"
            favorites={
              Array [
                "concat",
                "charCodeAt",
              ]
            }
            method="stringMethods"
            setFavorite={[MockFunction]}
          />
        </div>
      </div>,
      "className": "Slideshow",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="cards-slider-wrapper"
            style={
              Object {
                "transform": "translateX(-0%)",
              }
            }
          >
            <Card
              cardNumber={0}
              currentPage="string"
              favorites={
                Array [
                  "concat",
                  "charCodeAt",
                ]
              }
              method="stringMethods"
              setFavorite={[MockFunction]}
            />
          </div>,
          false,
        ],
        "className": "cards-slider active-slide-0",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                cardNumber={0}
                currentPage="string"
                favorites={
                  Array [
                    "concat",
                    "charCodeAt",
                  ]
                }
                method="stringMethods"
                setFavorite={[MockFunction]}
              />,
            ],
            "className": "cards-slider-wrapper",
            "style": Object {
              "transform": "translateX(-0%)",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "function (item, index) {
        if (typeof item === 'number' ||
            typeof item === 'string') {
            return index ? \\"idx-\\" + index : \\"val-\\" + item;
        }
        if (!map.has(item)) {
            map.set(item, counter++);
            return uid(item);
        }
        return 'uid' + map.get(item);
    }",
              "nodeType": "class",
              "props": Object {
                "cardNumber": 0,
                "currentPage": "string",
                "favorites": Array [
                  "concat",
                  "charCodeAt",
                ],
                "method": "stringMethods",
                "setFavorite": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cards-slider active-slide-0"
        >
          <div
            className="cards-slider-wrapper"
            style={
              Object {
                "transform": "translateX(-0%)",
              }
            }
          >
            <Card
              cardNumber={0}
              currentPage="string"
              favorites={
                Array [
                  "concat",
                  "charCodeAt",
                ]
              }
              method="stringMethods"
              setFavorite={[MockFunction]}
            />
          </div>
        </div>,
        "className": "Slideshow",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <div
              className="cards-slider-wrapper"
              style={
                Object {
                  "transform": "translateX(-0%)",
                }
              }
            >
              <Card
                cardNumber={0}
                currentPage="string"
                favorites={
                  Array [
                    "concat",
                    "charCodeAt",
                  ]
                }
                method="stringMethods"
                setFavorite={[MockFunction]}
              />
            </div>,
            false,
          ],
          "className": "cards-slider active-slide-0",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Card
                  cardNumber={0}
                  currentPage="string"
                  favorites={
                    Array [
                      "concat",
                      "charCodeAt",
                    ]
                  }
                  method="stringMethods"
                  setFavorite={[MockFunction]}
                />,
              ],
              "className": "cards-slider-wrapper",
              "style": Object {
                "transform": "translateX(-0%)",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "function (item, index) {
        if (typeof item === 'number' ||
            typeof item === 'string') {
            return index ? \\"idx-\\" + index : \\"val-\\" + item;
        }
        if (!map.has(item)) {
            map.set(item, counter++);
            return uid(item);
        }
        return 'uid' + map.get(item);
    }",
                "nodeType": "class",
                "props": Object {
                  "cardNumber": 0,
                  "currentPage": "string",
                  "favorites": Array [
                    "concat",
                    "charCodeAt",
                  ],
                  "method": "stringMethods",
                  "setFavorite": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
